### PR TITLE
Add RWMutex for podLastSyncTime map

### LINKED ISSUE #1882: fatal error: concurrent map writes
**What happened**:
When edgecore runs for a long time, there will be a `concurrent map writes` panic 
image

**What you expected to happen**:
edgecore run well.
**How to reproduce it (as minimally and precisely as possible)**:
edgecore run for a long time
**Anything else we need to know?**:

**Environment**:
- KubeEdge version(e.g. `cloudcore/edgecore --version`):  v1.3.0

### PR DESCRIPTION
**Pull request by XJangel**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change

/kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind test
> /kind failing-test
> /kind feature

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #1882

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: edge/pkg/edged/edged.go [hunk 1]
```diff
@@ -162,6 +162,12 @@ type podReady struct {
 	podReadyLock sync.RWMutex
 }
 
+// podLastSyncTime holds the podLastSyncTime map and its lock
+type podLastSyncTime struct {
+	lastSyncTime        map[types.UID]time.Time
+	podLastSyncTimeLock *sync.RWMutex
```
Comment (fisherxu) [inline edge/pkg/edged/edged.go:168]: No need use pointer here

File: edge/pkg/edged/edged.go [hunk 2]
```diff
@@ -913,13 +920,14 @@ func (e *edged) consumePodAddition(namespacedName *types.NamespacedName) error {
...
 	podUID := pod.GetUID()
-	curPodStatus, err := e.podCache.GetNewerThan(podUID, e.podLastSyncTime[podUID])
+	curPodStatus, err := e.podCache.GetNewerThan(podUID, e.lastSyncTime[podUID])
 	if err != nil {
 		klog.Errorf("pod %s cache newer failed: %v", podName, err)
 		return err
 	}
 	result := e.containerRuntime.SyncPod(pod, curPodStatus, secrets, e.podAdditionBackoff)
-	e.podLastSyncTime[podUID] = time.Now()
+	e.lastSyncTime[podUID] = time.Now()
```
Comment (fisherxu) [inline edge/pkg/edged/edged.go:929]: Remove this line

**Comment by XJangel** [comment]
> thanks for the fixing, but all Get/Delete/Write operations need add lock, you can add some get/delete funcs like set.

I get it.

File: edge/pkg/edged/edged.go (hunk 2 above)
Comment (XJangel) [reply edge/pkg/edged/edged.go:929]: > Remove this line

Here should call set function. I make a mistake.

**Comment by GsssC** [comment]
@fisherxu @XJangel how about changing it to sync.Map

**Comment by XJangel** [comment]
> @fisherxu @XJangel how about changing it to sync.Map

**Comment by XJangel** [comment]
> @fisherxu @XJangel how about changing it to sync.Map

I agree.

File: edge/pkg/edged/edged.go [hunk 3]
```diff
@@ -913,13 +913,17 @@ func (e *edged) consumePodAddition(namespacedName *types.NamespacedName) error {
 	}
 
 	podUID := pod.GetUID()
-	curPodStatus, err := e.podCache.GetNewerThan(podUID, e.podLastSyncTime[podUID])
+	t, ok := e.podLastSyncTime.Load(podUID)
+	if !ok {
+		t = new(time.Time)
```
Comment (fisherxu) [inline edge/pkg/edged/edged.go:918]: t should not be the pointer, t = time.time{}

File: edge/pkg/edged/edged.go [hunk 4]
```diff
@@ -420,7 +420,7 @@ func newEdged(enable bool) (*edged, error) {
 		nodeIP:                    net.ParseIP(edgedconfig.Config.NodeIP),
 		recorder:                  recorder,
 		enable:                    enable,
-		podLastSyncTime:           make(map[types.UID]time.Time),
+		podLastSyncTime:           sync.Map{},
```
Comment (fisherxu) [inline edge/pkg/edged/edged.go:423]: sync.Map can also work without this line :)

File: edge/pkg/edged/edged.go (hunk 3 above)
Comment (XJangel) [reply edge/pkg/edged/edged.go:918]: > t should not be the pointer, t = time.time{}

Got it.

**Pull request merged**